internal/tofu/context_import_test.go

```diff
@@ -115,6 +119,228 @@ resource "aws_instance" "foo" {
+			t.Fatal("no instance with key \"foo\" in final state")
+		}
+		if got, want := instanceState.ProviderKey, addrs.StringKey("a"); got != want {
+			t.Errorf("wrong provider key %s; want %s", got, want)
+		}
+		if instanceState.Current == nil {
+			t.Fatal("final resource instance has no current object")
+		}
+
+		gotObjState, err := instanceState.Current.Decode(resourceTypeSchema.Block.ImpliedType())
+		if err != nil {
+			t.Fatalf("failed to decode final resource instance object state: %s", err)
+		}
+		wantObjState := &states.ResourceInstanceObject{
+			Value: cty.ObjectVal(map[string]cty.Value{
+				"id":             cty.StringVal("fake-import-id"),
+				"import_marker":  cty.StringVal("a"),
+				"refresh_marker": cty.StringVal("a"),
+			}),
+			Status:       states.ObjectReady,
+			Dependencies: []addrs.ConfigResource{},
+		}
+		if diff := cmp.Diff(wantObjState, gotObjState, ctydebug.CmpOptions); diff != "" {
+			t.Error("wrong final object state\n" + diff)
+		}
+	}
+
+	// Now we'll try importing into an instance that isn't declared, which should fail.
+	{
+		missingInstanceAddr := addrs.RootModuleInstance.ResourceInstance(
+			addrs.ManagedResourceMode, "test_thing", "test", addrs.StringKey("nonexist"),
+		)
+		t.Logf("importing into %s, which should fail because it's not configured", missingInstanceAddr)
+		log.Printf("[TRACE] TestContextImport_multiInstanceProviderConfig: importing into %s, which should fail because it's not configured", missingInstanceAddr)
+		_, diags := ctx.Import(m, states.NewState(), &ImportOpts{
+			Targets: []*ImportTarget{
+				{
+					CommandLineImportTarget: &CommandLineImportTarget{
+						Addr: missingInstanceAddr,
+						ID:   "another-fake-import-id",
+					},
+				},
+			},
+		})
+		// FIXME: This does not currently return any error, and instead just succeeds while doing
+		// absolutely nothing, because OpenTofu's validation of the given resource instance address
+		// actually happens in "package command" as part of the import command, and it only checks
+		// that there is a resource block associated with the given address, and not that there's
+		// an instance associated with it.
+		//
+		// Since we're skipping over that check here, the graph walk ends up noticing that
+		// test_thing.test["foo"] is configured and making a NodePlannableResourceInstance for it,
+		// but since there's no node for test_thing.test["nonexist"] at all (because it's in neither
+		// config nor state) the import request is just silently ignored.
+		//
+		// However, I think that quirk is already true for importing into undeclared instance keys
+		// even without a dynamic provider instance reference, so maybe we should just delete this
+		// test case and treat this as a general bug with the "tofu import" command to be fixed
+		// later as a separate project.
+		if !diags.HasErrors() {
+			t.Fatal("unexpected success; want failure importing into undeclared resource instance which therefore has no provider instance")
+		}
```

Comment:
This is actually me now reviewing myself :roll_eyes: but I guess I forgot to actually ask what you and the rest of the team thought we should do here: are we all comfortable with the tradeoff I described in the comment above? 

If so, I think that means just deleting this entire pseudo-subtest block because it's testing for something that isn't actually the expected behavior right now. I think we'd need to make broader fixes to how import works for this to actually fail as this test expects.

Reply:
I think we should not extend the scope of dynamic providers further so moving this bug into a separate issue (fix + tests) - is the preferred way to me.